### PR TITLE
luks support

### PR DESCRIPTION
**Pull request by lolyu**
This aims to provide support new image format luks and luks encryption
for qcow2 image. Currently, only limited functions are included:
1. image creation(snapshot)
2. vm creation
3. image check

Usage:
1. luks
"""
image_secret = redhat
image_format = luks
"""
2. luks-encrypted qcow2
"""
image_format = qcow2
encryption = luks
image_secret = redhat
"""

Signed-off-by: lolyu <lolyu@redhat.com>

### CODE REVIEW

**Comment by hellohellenmao** [comment]
Raw and qcow2 loops work normally with this patch. 

Qcow2: 
RESULTS    : PASS 109 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

Raw:
RESULTS    : PASS 45 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Comment by hellohellenmao** [comment]
Test installation with luks format. It works !!!

(1/1) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS

**Comment by lolyu** [comment]
```
 (1/1) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (2035.88 s)

```

**Comment by lolyu** [comment]
```
 (1/1) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.image_locking_read_test: PASS (303.70 s)
```

**Comment by lolyu** [comment]
@luckyh update, please take a review.

File: virttest/qemu_storage.py [hunk 1]
```diff
@@ -43,6 +173,74 @@ def __init__(self, params, root_dir, tag):
...
+    @property
+    def secret_objects(self):
+        """All secret objects str needed for command line."""
+        secret_objects = [secret for secret in
+                          getattr(self, "base_secrets", [])]
+        if self.secret:
+            secret_objects.append(self.secret)
+        return secret_objects
+
+    def store_secret(self):
```
Comment (luckyh) [inline virttest/qemu_storage.py]: Should be a member of `ImageSecret`, what do you think?

**Comment by lolyu** [comment]
1. installation and boot
```
 (1/3) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (2139.65 s)
 (2/3) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.rh_kernel_update: ERROR: 'list' object has no attribute 'splitlines' (102.18 s)
 (3/3) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.boot: PASS (49.89 s)
```

**Comment by hellohellenmao** [comment]
Installation for images with luks and luks-inside-qcow2 is successful. And the logs are ok to me.

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**